src/main/java/com/aws/greengrass/util/RootCAUtils.java

```diff
@@ -112,17 +115,20 @@ public static void downloadFileFromURL(String url, File f) throws IOException {
             }
 
             try (InputStream inputStream = executeResponse.responseBody().get();
-                 OutputStream outputStream = Files.newOutputStream(f.toPath(), StandardOpenOption.CREATE,
-                         StandardOpenOption.APPEND, StandardOpenOption.SYNC)) {
+                    OutputStream outputStream = Files.newOutputStream(f.toPath(), StandardOpenOption.CREATE,
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Path traversal vulnerability detected. User-controlled input in file paths
can allow attackers to access files outside intended directories using
`../` sequences. Secure your code by using framework functions like
`getCanonicalPath()`, `normalize()`, or validating paths with
`.startsWith()` checks. Learn more:
https://cwe.mitre.org/data/definitions/22.html